src/services/GroupService.js

```diff
@@ -428,43 +434,49 @@ async function getGroup(currentUser, groupId, criteria) {
     }
     return session
   }
-  const cache = await helper.getCacheInstance()
+
+  const redisClient = await helper.acquireRedisClient()
+  await redisClient.connect()
 
   let groupToReturn
 
   try {
     if (!criteria.skipCache) {
       // check for the availibility of the group in cache
-      groupToReturn = cache.get(groupId)
+      groupToReturn = JSON.parse(await redisClient.get(`Group:${groupId}`))
     }
 
     if (!criteria.skipCache && groupToReturn) {
+      logger.debug('group found in cache')
       if (!isAdmin) delete groupToReturn.status
 
       // if the group is private, the user needs to be a member of the group, or an admin
       if (groupToReturn.privateGroup && currentUser !== 'M2M' && !helper.hasAdminRole(currentUser)) {
-        const cachedGroupMembers = cache.get(`${groupId}-members`)
+        const cachedGroupMembers = JSON.parse(await redisClient.get(`GroupMembers:${groupId}`))
 
         if (!_.includes(cachedGroupMembers, currentUser.userId)) {
           await helper.ensureGroupMember(getSession(), groupId, currentUser.userId)
-
           cachedGroupMembers.push(currentUser.userId)
-          cache.set(`${groupId}-members`, cachedGroupMembers)
+          await redisClient.set(`GroupMembers:${groupId}`, JSON.stringify(cachedGroupMembers), { EX: config.CACHE_TTL, NX: true })
```

Comment:
Shouldn't NX be false here? We enter this block if skipCache = false. So any writes here means you _want_ to over write the cached value. If that's not the intent this set should be removed.

Reply:
You could set KEEPTTL to true if the intent is to update the data but retain the TTL, but I'd check docs to be 100% sure.